testkit/src/main/scala/org/bitcoins/testkit/core/gen/CreditingTxGen.scala

```diff
@@ -40,15 +48,34 @@ sealed abstract class CreditingTxGen {
     //note, cannot put a p2wpkh here
     Gen.oneOf(p2pkOutput,
               p2pkhOutput,
-              multiSigOutput, /*cltvOutput,*/ csvOutput)
+              multiSigOutput, /*cltvOutput,*/ csvOutput,
+              conditionalOutput)
   }
 
-  def nonP2SHOutput: Gen[BitcoinUTXOSpendingInfo] = {
-    Gen.oneOf(p2pkOutput,
-              p2pkhOutput,
-              multiSigOutput, /*cltvOutput,*/ csvOutput,
-              p2wpkhOutput,
-              p2wshOutput)
+  /** Only for use in constructing P2SH outputs */
+  private def nonP2SHOutput: Gen[BitcoinUTXOSpendingInfo] = {
+    Gen
+      .oneOf(p2pkOutput,
+             p2pkhOutput,
+             multiSigOutput, /*cltvOutput,*/ csvOutput,
+             conditionalOutput,
+             p2wpkhOutput,
+             p2wshOutput)
+      .suchThat(_.scriptPubKey.compactSizeUInt.toLong + CompactSizeUInt(
```

Comment:
1. I think this is actually incorrect, this is checking if the _entire script_ is less than 520 bytes, where as the consesus rule is that largest _single element_ that can be pushed onto the [stack is 520 bytes](https://github.com/bitcoin-s/bitcoin-s/blob/d86acfffedda99ad73f6997fbf750f873f7ab2d3/core/src/main/scala/org/bitcoins/core/script/interpreter/ScriptInterpreter.scala#L41). The largest _total script size_ is [10,000 bytes](https://github.com/bitcoin-s/bitcoin-s/blob/d86acfffedda99ad73f6997fbf750f873f7ab2d3/core/src/main/scala/org/bitcoins/core/script/interpreter/ScriptInterpreter.scala#L436). It probably makes sense to make a helper method in `BitcoinScriptUtil.isValidStackElementSize()` or something like that. 

2. Another thing I generally don't like to use `.suchThat()` as it tends to generate a lot of generator failures where we cannot create a satisfiable `T` (in this case `BitcoinUTXOSpendingInfo`). 

I guess in this case it might be ok though, as it is pretty rare to generate a single stack element in the SPK with > 520 bytes in size.

Reply:
Here is a stackoverflow post talking about what happens when you use `.suchThat()` too much: https://stackoverflow.com/questions/45304489/why-is-my-scalacheck-test-with-a-custom-generator-failing-after-discarding-many

Reply:
In response to 1, When it comes to `P2SH` we push the redeem script onto the stack, hence the limit being needed.

In response to 2, I agree that in most cases this would not be ideal but the probability is low enough that this should be okay, as you say.